otx/cli/manager/config_manager.py

```diff
@@ -0,0 +1,405 @@
+        hyper_parameters = self.template.hyper_parameters.data
+        type_hint = gen_param_help(hyper_parameters)
+        updated_hyper_parameters = gen_params_dict_from_args(self.args, type_hint=type_hint)
+        override_parameters(updated_hyper_parameters, hyper_parameters)
+        return create(hyper_parameters)
+
+    def get_dataset_config(self, subsets: List[str]) -> dict:
+        """Returns dataset_config in a format suitable for each subset.
+
+        Args:
+            subsets (list, str): Defaults to ["train", "val", "unlabeled"].
+
+        Returns:
+            dict: dataset_config
+        """
+        dataset_config = {"task_type": self.task_type}
+        for subset in subsets:
+            if f"{subset}_subset" in self.data_config and self.data_config[f"{subset}_subset"]["data_root"]:
+                dataset_config.update({f"{subset}_data_roots": self.data_config[f"{subset}_subset"]["data_root"]})
+        return dataset_config
+
+    def update_data_config(self, data_yaml: dict) -> None:
+        # TODO: This also requires uniformity in the format.
+        """Convert the data yaml format to the data_config format consumed by the task.
+
+        Args:
+            data_yaml (dict): data.yaml format
+        """
+        if data_yaml["data"]["train"]["data-roots"]:
+            self.data_config["train_subset"] = {"data_root": data_yaml["data"]["train"]["data-roots"]}
+        if data_yaml["data"]["val"]["data-roots"]:
+            self.data_config["val_subset"] = {"data_root": data_yaml["data"]["val"]["data-roots"]}
+        if data_yaml["data"]["test"]["data-roots"]:
+            self.data_config["test_subset"] = {"data_root": data_yaml["data"]["test"]["data-roots"]}
+        if "unlabeled" in data_yaml["data"] and data_yaml["data"]["unlabeled"]["data-roots"]:
+            self.data_config["unlabeled_subset"] = {
+                "data_root": data_yaml["data"]["unlabeled"]["data-roots"],
+                "file_list": data_yaml["data"]["unlabeled"]["file-list"],
+            }
+        # FIXME: Hardcoded for Self-Supervised Learning
+        if self.mode == "train" and str(self.train_type).upper() == "SELFSUPERVISED":
+            self.data_config["val_subset"] = {"data_root": None}
+
+    def _get_template(self, task_type: str, model: str = None) -> ModelTemplate:
+        """Returns the appropriate template for each situation.
+
+        Args:
+            task_type (str): The task_type registered in the registry. Used for filtering.
+            model (str, optional): The task_type registered in the registry. Used for filtering. Defaults to None.
+
+        Returns:
+            ModelTemplate: Selected model template.
+        """
+        otx_registry = OTXRegistry(self.otx_root).filter(task_type=task_type)
+        if model:
+            template_lst = [temp for temp in otx_registry.templates if temp.name.lower() == model.lower()]
+            if len(template_lst) == 0:
```

Comment:
```suggestion
            if not template_lst:
```

It's recommended way to check empty list in [PEP 8 style guide](https://www.python.org/dev/peps/pep-0008).

Reply:
reflected this. https://github.com/openvinotoolkit/training_extensions/pull/1641/commits/3fcf067f3ae5f70826b0df66bdb702db0a04c358